File: haskell/hamming/src/Hamming.hs
module Hamming (distance) where

distance :: String -> String -> Maybe Int
distance xs ys =
    if length xs == length ys 
    then Just $ sum $ zipWith (\c1 c2 -> if c1 == c2 then 0 else 1) 
                              xs ys
    else Nothing

File: csharp/hamming/Hamming.cs
using System;
using System.Linq;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if (firstStrand.Length != secondStrand.Length)
            throw new ArgumentException("Strands must have equal length.");

        return firstStrand.Zip(secondStrand, (c1, c2) => c1 != c2 ? 1 : 0).Sum();
    }
}